### PR TITLE
etcdserver: Move get/put/delete on Lease bucket to bucket package

### PR DESCRIPTION
**Pull request by serathius**
cc @ptabor

### CODE REVIEW

File: server/mvcc/buckets/lease.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	err := lpb.Unmarshal(vs[0])
+	if err != nil {
+		panic("failed to unmarshal lease proto item")
+	}
+	return lpb
+}
+
+func int64ToBytes(n int64) []byte {
+	bytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(bytes, uint64(n))
```
Comment (ptabor) [inline server/mvcc/buckets/lease.go:71]: This conversion is very opinionated (bigendian & after casting to uint64) ... 

shell we name it more 'unique':   leaseIdToBytes(...) ?

File: server/mvcc/buckets/lease.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		panic("failed to marshal lease proto item")
+	}
+	tx.UnsafePut(Lease, key, val)
+}
+
+func UnsafeDeleteLease(tx backend.BatchTx, lpb *leasepb.Lease) {
+	tx.UnsafeDelete(Lease, int64ToBytes(lpb.ID))
+}
+
+func MustUnsafeGetLease(tx backend.BatchTx, lpb *leasepb.Lease) *leasepb.Lease {
```
Comment (ptabor) [inline server/mvcc/buckets/lease.go:57]: If its not crucial for performance (I assume so), pls make the method take only LeaseId and return the lpb. 

Alternatively (not recommended): Please add a comment that it mutates `lpb`.

**Review by ptabor — CHANGES_REQUESTED** [review]

File: server/mvcc/buckets/lease.go (hunk 2 above)
Comment (serathius) [reply server/mvcc/buckets/lease.go:57]: Using leasepb here is not great, but I proposed it as a temporary solution. Using LeaseId here is not possible as it creates import cycle (lease module defines LeaseID, but also imports MustUnsafeGetLease from buckets, buckets need to import LeaseID). I will try a not recommended option of adding comment, and prepare a separate PR with proper solution.

The proper solution would be to have buckets (should be renamed to schema at that point) define a struct that consumes LeaseID. In lease module we will define an interface LeaseStorage based on the struct methods. As this struct will be injected to interface, lease module will no longer import buckets directly.

Comment (ptabor) [reply server/mvcc/buckets/lease.go:58]: `int64 leaseId` is also a mid-ground option.

**Review by ptabor — APPROVED** [review]

File: server/mvcc/buckets/lease.go (hunk 2 above)
Comment (serathius) [reply server/mvcc/buckets/lease.go:58]: Done.

**Pull request merged**